### ISSUE TITLE
Allow roles to be executed against the IP

### ISSUE THREAD

**Issue by karras** [theforeman/foreman_ansible#53]
It would be cool if there's a setting or similar, which allows one to target the IP address instead of the hostname when executing the Ansible playbook. In my case users also provision VMs on their devices and therefore have no DNS record assigned (DHCP is not managed by Foreman).

In such a case the Foreman tasks will fail because the hostname will not be resolvable.

**Comment by dLobatog** [comment]
Good point. I guess we could use the primary interface of the Host for that, all compute resources that provision without DNS (like EC2, Azure, etc...) will  set the primary interface as the public IP coming from the compute resource.

**Comment by chilicat** [comment]
+1

**Comment by freysong** [comment]
+1

**Comment by zhuanjiaozhou** [comment]
+1

**Comment by sadoMasupilami** [comment]
+1

**Comment by selmison** [comment]
+1

**Comment by ares** [comment]
This is actually implemented by last commit in https://github.com/theforeman/foreman_ansible/pull/117
